server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetAliasesAction.java

```diff
@@ -94,51 +91,68 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         return channel -> client.admin().indices().getAliases(getAliasesRequest, new RestBuilderListener<GetAliasesResponse>(channel) {
             @Override
             public RestResponse buildResponse(GetAliasesResponse response, XContentBuilder builder) throws Exception {
-                final ImmutableOpenMap<String, List<AliasMetaData>> aliasMap = response.getAliases();
-
-                final Set<String> aliasNames = new HashSet<>();
                 final Set<String> indicesToDisplay = new HashSet<>();
-                for (final ObjectObjectCursor<String, List<AliasMetaData>> cursor : aliasMap) {
+                final Set<String> returnedAliasNames = new HashSet<>();
+                for (final ObjectObjectCursor<String, List<AliasMetaData>> cursor : response.getAliases()) {
                     for (final AliasMetaData aliasMetaData : cursor.value) {
-                        aliasNames.add(aliasMetaData.alias());
                         if (namesProvided) {
+                            // only display indices that have aliases
                             indicesToDisplay.add(cursor.key);
                         }
+                        returnedAliasNames.add(aliasMetaData.alias());
                     }
                 }
-
-                // first remove requested aliases that are exact matches
-                final SortedSet<String> difference = Sets.sortedDifference(Arrays.stream(aliases).collect(Collectors.toSet()), aliasNames);
-
-                // now remove requested aliases that contain wildcards that are simple matches
-                final List<String> matches = new ArrayList<>();
-                outer:
-                for (final String pattern : difference) {
-                    if (pattern.contains("*")) {
-                        for (final String aliasName : aliasNames) {
-                            if (Regex.simpleMatch(pattern, aliasName)) {
-                                matches.add(pattern);
-                                continue outer;
+                // compute explicitly requested aliases that have are not returned in the result
```

Comment:
it would be great to have a unit test for this code. Otherwise we only rely on yaml tests to test it. Could be done as a followup.